src/Util/Guid.php

```diff
@@ -33,7 +33,9 @@ public static function generateString(): string
 
     public static function generateAsHex(): string
     {
-        return self::generate()->getHex();
+        // string cast is a BC layer for gedmo/uuid v3
+        // to be refactor when dropping retrocompatibility
+        return (string) self::generate()->getHex();
```

Comment:
why not simply `return self::generate()->getHex()->toString();` ?

Reply:
Well, that's actually more complicated lol.

But the why is because the point of this is to maintain BC with old v3 ramsey/uuid which returns a string (and `string->toString()` will obviously fail)